src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

```diff
@@ -187,6 +173,10 @@ private async Task DoReceive()
             {
                 // If Shutdown() has already bee called, assume that was the reason ProcessReceives() exited.
                 Input.Complete(_shutdownReason ?? error);
+
+                FireConnectionClosed();
```

Comment:
This makes me wish we called to ConnectionClosed token FinReceived or something like that, because the connection might be only half closed at this point.

Reply:
You could argue the same about HttpContext.ConnectionAborted, I don't think it makes a difference.

Reply:
You absolutely could, but at least that's HTTP-specific so it makes a bit more sense.